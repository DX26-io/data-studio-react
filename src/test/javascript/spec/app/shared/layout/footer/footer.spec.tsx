import React from 'react';
import { shallow } from 'enzyme';

import {View, Flex, Text} from '@adobe/react-spectrum'

import Footer from 'app/shared/layout/footer/footer';


describe('Footer', () => {
  let mountedWrapper

  const defaultProps = {}

  const wrapper = (props = defaultProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<Footer {...props}/>)
    }
    return mountedWrapper
  };

  beforeEach(() => {
    mountedWrapper = undefined
  });

  it('Renders Footer with default Props', () => {
    const component = wrapper()
    expect(component).toMatchSnapshot()

    const mainView = component.find(View)
    expect(mainView.length).toEqual(1)

    const flex = mainView.find(Flex)
    expect(mainView.length).toEqual(1)

    const text = flex.find(Text)
    expect(text.length).toEqual(1)
  })
});
