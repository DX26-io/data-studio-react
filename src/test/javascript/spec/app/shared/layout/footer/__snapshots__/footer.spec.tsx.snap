// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Renders Footer with default Props 1`] = `
<ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)
  height="size-400"
  paddingX="size-150"
  paddingY="size-75"
>
  <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
    alignItems="center"
    justifyContent="end"
  >
    <ForwardRef($fe1955053990ab873941970c7463$var$Text)>
      Dx26
       
      Â©
       2021
    </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
  </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
</ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)>
`;
